File: cpp_brains/sigmoid.cpp
#include "sigmoid.h"

#include <cmath>

Sigmoid::~Sigmoid()
{

}

double Sigmoid::activate(double x) const
{
    return 1.0 / (1 + std::exp(-x));
}

double Sigmoid::derivative(double x) const
{
    double a = activate(x);
    return a * (1 - a);
}

File: kotlin_brians/src/main/kotlin/activations/Sigmoid.kt
package activations

import kotlin.math.exp

class Sigmoid : Activation() {
    override fun activate(x: Double): Double {
        return 1.0 / (1 + exp(-x))
    }

    override fun derivative(x: Double): Double {
        val a = activate(x)
        return a * (1 - a)
    }
}